lib/site-inspector.rb

```diff
@@ -127,37 +134,213 @@ def to_json
     to_hash.to_json
   end
 
-  def to_hash
+  def http
+    details = {
+      # site-inspector's best guess
+      canonical: uri.to_s,
+      canonical_protocol: https? ? :https : :http,
+      canonical_endpoint: www? ? :www : :root,
+
+      endpoints: endpoints
+    }
+
+    combos = details[:endpoints]
+
+    details[:live] = (
+      combos[:https][:www][:live] or
+      combos[:https][:root][:live] or
+      combos[:http][:www][:live] or
+      combos[:http][:root][:live]
+    )
+
+    details[:broken_root] = (
+      (combos[:https][:root][:status] == 0) and
+      (combos[:http][:root][:status] == 0)
+    )
+
+    details[:broken_www] = (
+      (combos[:https][:www][:status] == 0) and
+      (combos[:http][:www][:status] == 0)
+    )
+
+    details[:enforce_https] = (
+      (
+        (combos[:http][:www][:status] == 0) ||
+        (combos[:http][:www][:redirect_to_https])
+      ) and
+      (
+        (combos[:http][:root][:status] == 0) or
+        (combos[:http][:root][:redirect_to_https])
+      ) and
+      (
+        combos[:https][:www][:live] or
+        combos[:https][:root][:live]
+      )
+    )
```

Comment:
HTTPS is considered "enforced" for a domain if:

* Both `http://` endpoints are either inaccessible, or redirect eventually to an `https://` URL.
* At least one of the `https://` endpoints is "live" and eventually leads to a 200 request.

It's possible a domain's HTTPS endpoint could redirect to some other domain or subdomain's HTTP endpoint -- but that would still mean that the domain itself "enforces HTTPS". 

Whether or not a domain enforces HTTPS for all of its subdomains can only be measured by running this individually on each subdomain, or by evaluating the HSTS header on the base domain.